File: crossecore-java/src/java_/EnumDeclarationBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class EnumDeclarationBase 
extends AbstractTypeDeclarationImpl implements EnumDeclaration
{
	private com.crossecore.ocl.OrderedSet<EnumConstantDeclaration> _enumConstants;
	
	public com.crossecore.ocl.OrderedSet<EnumConstantDeclaration> getEnumConstants()
	{
		if(_enumConstants==null){
			_enumConstants = new com.crossecore.ocl.OrderedSet<EnumConstantDeclaration>(EnumConstantDeclaration.class, this, Java_PackageImpl.ENUM_DECLARATION__ENUM_CONSTANTS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.ENUM_DECLARATION__ENUM_CONSTANTS);
		}
		return _enumConstants;
	
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.ENUMDECLARATION;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.ENUM_DECLARATION__ENUM_CONSTANTS:
				return getEnumConstants();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.ENUM_DECLARATION__ENUM_CONSTANTS:
				getEnumConstants().clear();
				getEnumConstants().addAll((java.util.Collection<? extends EnumConstantDeclaration>) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.ENUM_DECLARATION__ENUM_CONSTANTS:
				return _enumConstants != null && !_enumConstants.isEmpty();
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/EnumDeclarationBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class EnumDeclarationBase 
	:AbstractTypeDeclarationImpl, EnumDeclaration
	{
		private OrderedSet<EnumConstantDeclaration> _enumConstants;
		
		public virtual OrderedSet<EnumConstantDeclaration> enumConstants
		{
			get {
				if(_enumConstants==null){
					_enumConstants = new OrderedSet<EnumConstantDeclaration>(this, Java_PackageImpl.ENUMDECLARATION_ENUMCONSTANTS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.ENUMDECLARATION_ENUMCONSTANTS);
				}
				return _enumConstants;
			}
		
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.ENUMDECLARATION;
		}
		
		
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.ENUMDECLARATION_ENUMCONSTANTS:
					return enumConstants;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.ENUMDECLARATION_ENUMCONSTANTS:
					enumConstants.Clear();
					enumConstants.AddRange(((List<EObject>)newValue)?.Cast<EnumConstantDeclaration>());
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.ENUMDECLARATION_ENUMCONSTANTS:
					return _enumConstants != null && !_enumConstants.isEmpty();
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
